app/models/manageiq/providers/redhat/infra_manager/refresh/strategies/api4.rb

```diff
@@ -1,4 +1,59 @@
 module ManageIQ::Providers::Redhat::InfraManager::Refresh::Strategies
   class Api4 < ManageIQ::Providers::Redhat::InfraManager::Refresh::Refresher
+    attr_reader :ems
+
+    def inventory_from_rhv(ems)
+      @ems = ems
+      InventoryWrapper.new(old_inventory: super, ems: ems)
+    end
+
+    class InventoryWrapper
+      attr_reader :old_inventory
```

Comment:
Not sure how this works, but the fact that you mention `old_...` makes me think that you are trying to adapt the V4 inventory to the V3 one. We should try to avoid that, if adaptations are required they should go in the V3 code, which is the one that we want to eventually remove.